src/test/java/org/broadinstitute/consent/http/service/DatasetRegistrationServiceTest.java

```diff
@@ -0,0 +1,531 @@
+        schemaV1.setSpecies(RandomStringUtils.randomAlphabetic(10));
+        schemaV1.setPiName(RandomStringUtils.randomAlphabetic(10));
+        when(user.getUserId()).thenReturn(1);
+        schemaV1.setDataSubmitterUserId(user.getUserId());
+        schemaV1.setDataCustodianEmail(List.of(RandomStringUtils.randomAlphabetic(10)+"@domain.org"));
+        schemaV1.setPublicVisibility(true);
+        schemaV1.setDataAccessCommitteeId(1);
+
+        ConsentGroup consentGroup1 = new ConsentGroup();
+        consentGroup1.setConsentGroupName(RandomStringUtils.randomAlphabetic(10));
+        consentGroup1.setGeneralResearchUse(true);
+        FileTypeObject fileType1 = new FileTypeObject();
+        fileType1.setFileType(FileTypeObject.FileType.Arrays);
+        fileType1.setFunctionalEquivalence(RandomStringUtils.randomAlphabetic(10));
+        fileType1.setNumberOfParticipants(new Random().nextInt());
+        consentGroup1.setFileTypes(List.of(fileType1));
+
+        ConsentGroup consentGroup2 = new ConsentGroup();
+        consentGroup1.setConsentGroupName(RandomStringUtils.randomAlphabetic(10));
+        consentGroup1.setGeneralResearchUse(true);
+        FileTypeObject fileType2 = new FileTypeObject();
+        fileType2.setFileType(FileTypeObject.FileType.Arrays);
+        fileType2.setFunctionalEquivalence(RandomStringUtils.randomAlphabetic(10));
+        fileType2.setNumberOfParticipants(new Random().nextInt());
+        consentGroup1.setFileTypes(List.of(fileType2));
+
+
+        schemaV1.setConsentGroups(List.of(consentGroup1, consentGroup2));
+        return schemaV1;
+    }
+
+
+    private DatasetRegistrationSchemaV1 createRandomCompleteDatasetRegistration(User user) {
+        DatasetRegistrationSchemaV1 schemaV1 = new DatasetRegistrationSchemaV1();
+        schemaV1.setStudyName(RandomStringUtils.randomAlphabetic(10));
+        schemaV1.setStudyType(DatasetRegistrationSchemaV1.StudyType.Observational);
+        schemaV1.setStudyDescription(RandomStringUtils.randomAlphabetic(10));
+        schemaV1.setDataTypes(List.of(RandomStringUtils.randomAlphabetic(10)));
+        schemaV1.setPhenotypeIndication(RandomStringUtils.randomAlphabetic(10));
+        schemaV1.setSpecies(RandomStringUtils.randomAlphabetic(10));
+        schemaV1.setPiName(RandomStringUtils.randomAlphabetic(10));
+        when(user.getUserId()).thenReturn(1);
+        schemaV1.setDataSubmitterUserId(user.getUserId());
+        schemaV1.setDataCustodianEmail(List.of(RandomStringUtils.randomAlphabetic(10)+"@domain.org"));
+        schemaV1.setPublicVisibility(true);
+        schemaV1.setDataAccessCommitteeId(1);
+        schemaV1.setSubmittingToAnvil(true);
+        schemaV1.setDbGaPPhsID(RandomStringUtils.randomAlphabetic(10));
+        schemaV1.setDbGaPStudyRegistrationName(RandomStringUtils.randomAlphabetic(10));
+        schemaV1.setEmbargoReleaseDate("2007-12-03");
+        schemaV1.setSequencingCenter(RandomStringUtils.randomAlphabetic(10));
```

Comment:
I wonder if there's a way to improve creating random fields without having to manually set all the fields like this.

Reply:
I agree - I'm not sure there is a great way, though. A similar pattern can be seen throughout the app, but this is just one of the bigger objects.

Reply:
Perhaps worth adding a TODO comment here.